src/models/beast.ts

```diff
@@ -1,35 +1,133 @@
+ */
 export default class BeastModel {
+    /**
+     * Create a new beast with a unique gamerTag and email
+     * @param gamerTag Unique gamerTag (username)
+     * @param email Unique email
+     */
     public static async create(gamerTag: string, email: string): Promise<Beast> {
         try {
-            const beast = await prisma.beast.create({
+            return await prisma.beast.create({
                 data: {
                     gamerTag,
                     email,
                 },
             });
+        } catch (error: unknown) {
+            BeastModel.handleErrors(error);
+        }
+        // this won't ever get hit, but eslint doesn't like when its not here for some reason lol
+        throw new UnknownError('An unexpected error occurred');
+    }
+
+    /**
+     * Get a beast by its id, also includes the beast's friends and preeshes made/received
+     * @param id Id of the requested beast
+     */
+    public static async getBeastById(id: number): Promise<Beast | null> {
+        try {
+            const beast = await prisma.beast.findUnique({
+                where: {
+                    id,
+                },
+                include: {
+                    beastieBros: true,
+                    authoredPreesh: {
+                        include: {
+                            comments: true,
+                        },
+                    },
+                    receivedPreesh: {
+                        include: {
+                            comments: true,
+                        },
+                    },
+                },
+            });
+            if (!beast) {
+                throw new NotFoundError(`Beast with id: ${id} does not exist.`);
+            }
 
             return beast;
         } catch (error: unknown) {
-            if (error instanceof Prisma.PrismaClientKnownRequestError) {
-                if (isErrorWithTarget(error)) {
-                    handleDatabaseError(error);
-                } else {
-                    throw new UnknownError(error.message);
-                }
-            } else if (error instanceof Prisma.PrismaClientValidationError) {
-                throw new UnprocessableEntityError(error.message);
+            BeastModel.handleErrors(error);
+        }
+        throw new UnknownError('An unexpected error occurred');
+    }
+
+    /**
+     * Updates a beast with the given (optional) gamerTag and (optional) email
+     * @param id the id of the beast to update
+     * @param gamerTag the new gamerTag
+     * @param email the new email
+     */
+    public static async updateBeast(id: number, gamerTag?: string, email?: string): Promise<Beast | null> {
+        // Will only update with the fields that are provided
+        const updateData: { gamerTag?: string; email?: string; } = {};
+
+        if (gamerTag) {
+            updateData.gamerTag = gamerTag;
+        }
+
+        if (email) {
+            updateData.email = email;
+        }
```

Comment:
Instead of an if statement for each field why not just do:
const updateData = { gamerTag, email }
If they exist they will be in there, if not, they'll be undefined